x-pack/plugins/uptime/public/components/synthetics/check_steps/waterfall_marker_trend.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { EuiButton } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+import { useSelector } from 'react-redux';
+import { useUptimeStartPlugins } from '../../../contexts/uptime_startup_plugins_context';
+import { JourneyStep } from '../../../../common/runtime_types';
+import { AllSeries, createExploratoryViewUrl } from '../../../../../observability/public';
+import { euiStyled } from '../../../../../../../src/plugins/kibana_react/common';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
+import { selectDynamicSettings } from '../../../state/selectors';
+
+export const getLast48Intervals = (activeStep: JourneyStep) => {
```

Comment:
It would be nice to have unit tests around this function since it has quite a bit of logic and different possible outputs.

Reply:
Done in [`b3eab1b` (#116266)](https://github.com/elastic/kibana/pull/116266/commits/b3eab1be59729f4749290875a78094383a99d1af)